### ISSUE TITLE
Update rust-bitcoin

### ISSUE THREAD

**Issue by jkitman** [fedimint/fedimint#2356]
It's one of our main crypto libs we are a few versions behind.

edit (justinmoon): edited this to be "update rust-bitcoin" from "update secp256k1"

**Comment by elsirion** [comment]
This will have to happen in lock step with rust-bitcoin, miniscript, all the BTC clients using rust-bitcoin structs (bitcoind, esplora, electrum) and possibly some other crates. We need to make sure that all required ones support the same secp version.

**Comment by elsirion** [comment]
This will likely be part of #2667.

**Comment by m1sterc001guy** [comment]
This is also blocked on the `lightning` and `lightning-invoice` crates. They do not support `bitcoin 0.30.1` or `secp256k1 0.27.0` yet.

**Comment by benthecarman** [comment]
Latest `LDK` release is now on `0.30`

https://github.com/lightningdevkit/rust-lightning/releases/tag/v0.0.119

**Comment by EthnTuttle** [comment]
> Latest `LDK` release is now on `0.30`
> 
> https://github.com/lightningdevkit/rust-lightning/releases/tag/v0.0.119

https://github.com/lightningdevkit/rust-lightning/pull/2799

**Comment by justinmoon** [comment]
> [lightningdevkit/rust-lightning#2799](https://github.com/lightningdevkit/rust-lightning/pull/2799)

If this is the only regression, I think we can still integrate because the only part of LDK that currently runs in WASM is `lightning_invoice` and the regression this PR fixes is elsewhere. 

I propose we update on master but don't backport it.

**Comment by tvolk131** [comment]
If no one's currently working on this, I can give it a shot!

**Comment by m1sterc001guy** [comment]
I have a broken branch attempting to do this here: https://github.com/m1sterc001guy/fedimint/tree/update_rust_bitcoin

These updates touch a lot of files and some structs in `rust-bitcoin` removed their `Encodable` and `Decodable` traits.

**Comment by justinmoon** [comment]
Relevant TODO https://github.com/fedimint/fedimint/blob/47eb296afc274c9526163993141edc42f8c8c9d1/fedimint-core/src/session_outcome.rs#L70-L78

**Comment by tvolk131** [comment]
Unless anyone knows of more work to do for this issue, I think we can mark this as done!

**Comment by m1sterc001guy** [comment]
@tvolk131 is going to make a separate issue for removing bitcoin 29 from the codebase.

**Comment by benthecarman** [comment]
congrats!
